### PR TITLE
Show proper error for misusing the custom natspec tag

### PR DESCRIPTION
**Pull request by axic**
I think this is the real solution to #11152.

Made it two commits so the original error is easy to see, but it can be squashed prior to merge.

### CODE REVIEW

File: test/libsolidity/syntaxTests/natspec/invalid/invalid_tag.sol [hunk 1]
```diff
@@ -14,6 +14,6 @@ contract C {
 // ----
 // DocstringParsingError 6546: (0-14): Documentation tag @a&b not valid for contracts.
 // DocstringParsingError 2968: (28-49): Invalid character in custom tag @custom:x^y. Only lowercase letters and "-" are permitted.
-// DocstringParsingError 6546: (80-92): Documentation tag @custom: not valid for functions.
+// DocstringParsingError 6564: (80-92): Custom documentation tag must contain a chosen name, i.e. @custom:mytag.
```
Comment (axic) [inline test/libsolidity/syntaxTests/natspec/invalid/invalid_tag.sol:17]: I swear I used the error code script to generate the number, but somehow RNGesus decided 6546 -> 6564 is a good number.

**Review by Marenz — APPROVED** [review]

**Review by hrkrshnn — COMMENTED** [review]
Missing changelog.

**Comment by axic** [comment]
@hrkrshnn it is just an error reporting change, do we need it? I'm fine to add it.

Any suggestions for the error message? Does it look confusing that the `.` is part of the tag?

**Review by hrkrshnn — APPROVED** [review]
On second thought, maybe the changelog is not too important.

**Review by chriseth — APPROVED** [review]

**Pull request merged**